pkg/apis/core/validation/validation.go

```diff
@@ -4921,6 +4927,12 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 		}
 	}
 
+	if sc.RunAsGroup != nil {
+		if validation.IsValidGroupID(*sc.RunAsGroup) != nil {
```

Comment:
The error handling here should match the error handling in the pod version - loop over the error messages.